### ISSUE TITLE
About construction dictionary.

### ISSUE THREAD

**Issue by q-rong** [hobincar/RecNet#25]
Hi,

I wonder why the vocabulary is constructed with only the caption of the training set? Instead of the entire dataset?

**Comment by hobincar** [comment]
In the captioning model, each word is embedded into a vector through a word embedding matrix. For the words that are not in the training set, their word embeddings will not be learned and random vectors will be used at the validation/test stage.
Instead, we can treat all words that are not appearing in the training set by defining a special word `<UNK>` to embed them into the same representation. Furthermore, we can learn the word embedding of `<UNK>` by mapping rare words that appear less than `N` times in the training data, which is often referred to as the "rare word elimination" technique.

**Comment by q-rong** [comment]
Hi,

I still don't understand how to construct a vocabulary using only the caption of the training set. Using only the training set to construct the vocabulary, it must be missing the words in the test set and validation set.

When generating descriptive sentences, searching for words in a vocabulary constructed using only the training set is bound to be inaccurate.

Am I right?

**Comment by hobincar** [comment]
Yes, you're right. The model cannot predict the word that was not in the training set. However, allowing the new word to appear in the test phase does not make sense, since it does not know what the novel word means (as the word is never seen in the training phase).
